### PR TITLE
refactor: programmatic modal

### LINKED ISSUE #6277: Programmatic Modal
## Specs
- replace [Buefy Programmatic Modal](https://buefy.org/documentation/modal/#programmatic) with Oruga [equivalent](https://oruga.io/components/modal.html)
- close all opened modals when one is opened

### PR DESCRIPTION
**Pull request by preschian**
**Thank you for your contribution** to the [KodaDot - One Stop Shop for Polkadot NFTs](https://kodadot.xyz).

👇 __ Let's make a quick check before the contribution.

## PR Type

- [ ] Bugfix
- [ ] Feature
- [x] Refactoring

## Needs QA check

- @kodadot/qa-guild please review

## Context

- [x] Closes #6277

#### Did your issue had any of the "$" label on it?

- [x] My DOT address: [Payout](https://canary.kodadot.xyz/dot/transfer/?target=1xjvRADwdJcnmUCLWerEHR4iGCT5EBTm4D4fzLLg4LcAC9p)

### CODE REVIEW

File: libs/ui/src/components/NeoModalExtend/plugin.js [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+import Modal from './NeoModalExtend.vue'
+
+import { merge } from '@oruga-ui/oruga/src/utils/helpers'
+import InstanceRegistry from '@oruga-ui/oruga/src/utils/InstanceRegistry'
+
+let instances = new InstanceRegistry()
+
+// fork https://github.com/oruga-ui/oruga/blob/4915c4dbcb1a6c2cd39bc7660d41916b330bc0cd/packages/oruga/src/components/modal/index.js#L12 implementation
+const ModalProgrammatic = {
+  open(params) {
```
[the remark this answers is not in the document]
Comment (preschian) [reply libs/ui/src/components/NeoModalExtend/plugin.js:11]: sorry. wont fix

File: components/Navbar.vue [hunk 2]
```diff
@@ -210,7 +209,7 @@ import { getChainNameByPrefix } from '@/utils/chain'
 import { createVisible, explorerVisible } from '@/utils/config/permision.config'
 import ShoppingCartButton from './navbar/ShoppingCartButton.vue'
 
-const { $buefy, $nextTick } = useNuxtApp()
+const { $nextTick, $neoModal } = useNuxtApp()
```
Comment (roiLeo) [inline components/Navbar.vue:212]: I think `nextTick` is auto imported with nuxt3

File: components/Navbar.vue [hunk 3]
```diff
@@ -248,15 +247,11 @@ const handleMobileChainSelect = () => {
...
+  $neoModal.open({
     parent: instance?.proxy,
     ...ConnectWalletModalConfig,
-  } as unknown as BModalConfig)
-
-  // close all modal
-  document.querySelectorAll('.modal').forEach((modal) => {
-    modal.__vue__?.$vnode?.context?.close()
-    modal.remove()
   })
```
Comment (roiLeo) [inline components/Navbar.vue:255]: usure

**Review by roiLeo — APPROVED** [review]
Nice! Even if we are going to rework this after migration.

Otherwise  I can't open shopping cart & notification on mobile (maybe it isn't related with this PR, works on beta)

**Review by Jarsen136 — APPROVED** [review]

**Comment by prury** [comment]
So far(Still testing): 

Navbar - 

drop:
DropContainer - ✔️ 
TokenImportButton

unlockable
UnlockableContainer.vue ✔️


Connect Wallet:
ConnectWalletModal.vue ✔️
useConnectWallet.ts ✔️

NotificationBox
useNotificationBox.ts


ShoppingCart ✔️
ShoppingCartModal.vue ✔️



GalleryItem
GalleryItemOffer.vue ✔️


Navbar:
NotificationBoxButton.vue
ProfileDropdown.vue
ShoppingCartButton.vue ✔️

redirect:
useRedirectModal.ts

RMRK/Gallery
EmotionList.vue
GalleryCard.vue


Shared:
ConnectWalletButton.vue

**Comment by yangwao** [comment]
pay 100 usd

**Comment by yangwao** [comment]
😍 Perfect, I’ve sent the payout 
💵 $100 @ 4.79 USD/DOT ~ 20.877 $DOT
🧗 [1xjvRADwdJcnmUCLWerEHR4iGCT5EBTm4D4fzLLg4LcAC9p](https://polkadot.subscan.io/account/1xjvRADwdJcnmUCLWerEHR4iGCT5EBTm4D4fzLLg4LcAC9p)
🔗 [0x5956899f54cc1ec1dae3ef1ada2fd096db173e30ab24bb2176b8992b4484ab8c](https://polkadot.subscan.io/extrinsic/0x5956899f54cc1ec1dae3ef1ada2fd096db173e30ab24bb2176b8992b4484ab8c)

🪅 Let’s grab another issue and get rewarded! 
🪄 [github.com/kodadot/nft-gallery/issues](https://github.com/kodadot/nft-gallery/issues/)

File: components/Navbar.vue (hunk 3 above)
Comment (preschian) [reply components/Navbar.vue:255]: 🤣

**Pull request merged**